diff --git a/node_modules/ssh2/lib/protocol/constants.js b/node_modules/ssh2/lib/protocol/constants.js
index ad77592..339ce87 100644
--- a/node_modules/ssh2/lib/protocol/constants.js
+++ b/node_modules/ssh2/lib/protocol/constants.js
@@ -78,7 +78,10 @@ const SUPPORTED_SERVER_HOST_KEY = DEFAULT_SERVER_HOST_KEY.concat([
 
 const canUseCipher = (() => {
   const ciphers = crypto.getCiphers();
-  return (name) => ciphers.includes(CIPHER_INFO[name].sslName);
+  return (name) => {
+    const info = CIPHER_INFO[name];
+    return !!info && ciphers.includes(info.sslName);
+  };
 })();
 let DEFAULT_CIPHER = [
   // http://tools.ietf.org/html/rfc5647
@@ -124,7 +127,10 @@ const SUPPORTED_CIPHER = DEFAULT_CIPHER.concat([
 
 const canUseMAC = (() => {
   const hashes = crypto.getHashes();
-  return (name) => hashes.includes(MAC_INFO[name].sslName);
+  return (name) => {
+    const info = MAC_INFO[name];
+    return !!info && hashes.includes(info.sslName);
+  };
 })();
 const DEFAULT_MAC = [
   'hmac-sha2-256-etm@openssh.com',
diff --git a/node_modules/ssh2/lib/protocol/crypto.js b/node_modules/ssh2/lib/protocol/crypto.js
index 851cc73..39141e4 100644
--- a/node_modules/ssh2/lib/protocol/crypto.js
+++ b/node_modules/ssh2/lib/protocol/crypto.js
@@ -1578,21 +1578,7 @@ module.exports = {
   MAC_INFO,
   bindingAvailable: !!binding,
   init: (() => {
-    // eslint-disable-next-line no-async-promise-executor
-    return new Promise(async (resolve, reject) => {
-      try {
-        POLY1305_WASM_MODULE = await require('./crypto/poly1305.js')();
-        POLY1305_RESULT_MALLOC = POLY1305_WASM_MODULE._malloc(16);
-        poly1305_auth = POLY1305_WASM_MODULE.cwrap(
-          'poly1305_auth',
-          null,
-          ['number', 'array', 'number', 'array', 'number', 'array']
-        );
-      } catch (ex) {
-        return reject(ex);
-      }
-      resolve();
-    });
+    return Promise.resolve();
   })(),
 
   NullCipher,
